libs/dal/src/repositories/topic/topic.schema.ts

```diff
@@ -0,0 +1,44 @@
+import * as mongoose from 'mongoose';
+import { Schema } from 'mongoose';
+
+import { TopicEntity } from './topic.entity';
+
+import { schemaOptions } from '../schema-default.options';
+
+const topicSchema = new Schema(
+  {
+    _environmentId: {
+      type: Schema.Types.ObjectId,
+      ref: 'Environment',
+      index: true,
+      required: true,
+    },
+    _organizationId: {
+      type: Schema.Types.ObjectId,
+      ref: 'Organization',
+      index: true,
+      required: true,
+    },
+    _userId: {
+      type: Schema.Types.ObjectId,
+      ref: 'User',
+      index: true,
+      required: true,
+    },
+    key: {
+      type: Schema.Types.String,
+      required: true,
+    },
+    name: {
+      type: Schema.Types.String,
+      required: true,
+    },
+    customData: {
+      type: Schema.Types.Mixed,
+    },
+  },
+  schemaOptions
+);
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const Topic = mongoose.models.Topic || mongoose.model<TopicEntity>('Topic', topicSchema);
```

Comment:
shouldn't this be like?

```suggestion
interface ITopicDocument extends TopicEntity, Document {
  _id: string;
}

export const Topic = mongoose.models.Topic || mongoose.model<ITopicDocument>('Topic', topicSchema);
```

Reply:
To be honest I do not understand what the Document interface adds and why it should be needed. Also I was having a clash because of the typings so decided to not use it.

Reply:
yeah, I'm not sure why we are doing it, and it seems to me like the types are used wrong, check this one: https://mongoosejs.com/docs/typescript.html

they want to drop the Document soon: 
`We strongly recommend against using this approach, its support will be dropped in the next major version as it causes major performance issues.`

Reply:
Let's try my implementation and if we find something weird we can always roll it back to what's been done in other entities. Many thanks for the article.